static/js/brand-store/Settings/Settings.test.js

```diff
@@ -15,59 +15,105 @@ jest.mock("react-router-dom", () => ({
+  setupMockSelector(initialState);
 
-  it("should render the Settings view", () => {
-    render(
-      <Provider store={store}>
-        <Settings />
-      </Provider>
-    );
-    expect(document.querySelector("#is_public")).toBeInTheDocument();
-  });
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
 
-  test("the 'is public' checkbox should not be checked when the current store is set to private", () => {
-    render(
-      <Provider store={store}>
-        <Settings />
-      </Provider>
-    );
-    expect(document.querySelector("#is_public").checked).toEqual(false);
-  });
+  expect(document.querySelector("#is_public").checked).toEqual(false);
+});
 
-  test("the 'is public' checkbox should be checked when the current store is not set to private", () => {
-    useSelector.mockImplementation((callback) => {
-      return callback({
-        currentStore: {
-          currentStore: {
-            id: "test",
-            private: false,
-            "manual-review-policy": "allow",
-          },
-        },
-      });
-    });
-
-    render(
-      <Provider store={store}>
-        <Settings />
-      </Provider>
-    );
-    expect(document.querySelector("#is_public").checked).toEqual(true);
-  });
+test("the 'is public' checkbox should be checked when the current store is not set to private", () => {
+  initialState.currentStore.currentStore.private = false;
+  setupMockSelector(initialState);
+
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
+
+  expect(document.querySelector("#is_public").checked).toEqual(true);
+});
+
+test("the correct value is given to the store ID field", () => {
+  setupMockSelector(initialState);
+
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
+
+  expect(screen.getByLabelText("Store ID").value).toEqual(
+    initialState.currentStore.currentStore.id
+  );
+});
+
+test("the correct radio button is checked by default for manual review policy", () => {
+  setupMockSelector(initialState);
+
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
+
+  const radioButtons = screen.getAllByRole("radio");
+  const checkedRadioButton = radioButtons.find(
+    (button) => button.checked === true
+  );
+
+  expect(checkedRadioButton.value).toEqual(
+    initialState.currentStore.currentStore["manual-review-policy"]
+  );
+});
+
+test("the save button is disabled by default", () => {
+  setupMockSelector(initialState);
+
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
+
+  expect(screen.getByText("Save changes").disabled).toEqual(true);
+});
+
+test("the save button is enabled when the data changes", () => {
+  setupMockSelector(initialState);
+
+  render(
+    <Provider store={store}>
+      <Settings />
+    </Provider>
+  );
+
+  fireEvent.click(screen.getByRole("checkbox"));
```

Comment:
It's preferential to use `userEvent` vs `fireEvent` however in this case you can probably just do...
```suggestion
  screen.getByRole("checkbox").click();
```